### PR TITLE
Dunfell yocto check layer changes + rebase + fix

### PR DESCRIPTION
**Pull request by lluiscampos**
See #1084

### CODE REVIEW

**Comment by lluiscampos** [comment]
Running tests together with https://github.com/mendersoftware/mender-qa/pull/416 @ [Pipeline 195242424](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195242424)

File: meta-mender-demo/recipes-core/psplash/psplash_git.bbappend [hunk 1]
```diff
@@ -4,4 +4,6 @@ SRC_URI_append_mender-client-install = " \
 	file://mender.io.png \
 "
 
-SPLASH_IMAGES_mender-client-install = "file://mender.io.png;outsuffix=default"
+# Explicitly set to all arch/machines, otherwise breaks on raspberrypi builds
+# with error "Nothing RPROVIDES 'psplash-raspberrypi'"
+SPLASH_IMAGES_mender-client-install_all = "file://mender.io.png;outsuffix=default"
```
Comment (kacf) [inline meta-mender-demo/recipes-core/psplash/psplash_git.bbappend:9]: To be honest I'm not 100% sure myself why this is needed. But can you try this instead?

```
SPLASH_IMAGES_append_mender-client-install = " file://mender.io.png;outsuffix=default"
```

(beware of the leading space inside the quotes!)

Not sure if it accomplishes the same thing as what you did, but it is more idiomatic.

Comment (lluiscampos) [reply meta-mender-demo/recipes-core/psplash/psplash_git.bbappend:9]: This works, too. I'll rework.

Comment (lluiscampos) [reply meta-mender-demo/recipes-core/psplash/psplash_git.bbappend]: Amended in [this commit](https://github.com/mendersoftware/meta-mender/pull/1097/commits/4fe19a1e02f951a8b376779c122fd4eeb7a9bfd4).

**Comment by lluiscampos** [comment]
> Running tests together with [mendersoftware/mender-qa#416](https://github.com/mendersoftware/mender-qa/pull/416) @ [Pipeline 195242424](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195242424)

New build @ [Pipeline 195338486](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195338486)

File: meta-mender-demo/recipes-core/psplash/psplash_git.bbappend (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-demo/recipes-core/psplash/psplash_git.bbappend]: @kacf Actually, it fixed raspberry builds but broke qemu ones. See this [build log](https://gitlab.com/Northern.tech/Mender/mender-qa/-/jobs/760702496), error follows:

```
NOTE: recipe psplash-0.1+gitAUTOINC+0a902f7cd8-r15: task do_populate_sysroot: Succeeded
ERROR: psplash-0.1+gitAUTOINC+0a902f7cd8-r15 do_package: QA Issue: psplash-default is listed in PACKAGES multiple times, this leads to packaging errors. [packages-list]
ERROR: psplash-0.1+gitAUTOINC+0a902f7cd8-r15 do_package: Fatal QA errors found, failing task.
NOTE: recipe psplash-0.1+gitAUTOINC+0a902f7cd8-r15: task do_package: Failed
ERROR: Logfile of failure stored in: /builds/Northern.tech/Mender/build-qemux86-64-uefi-grub/tmp/work/core2-64-poky-linux/psplash/0.1+gitAUTOINC+0a902f7cd8-r15/temp/log.do_package.8671
ERROR: Task (/builds/Northern.tech/Mender/meta/recipes-core/psplash/psplash_git.bb:do_package) failed with exit code '1'
```

Comment (kacf) [reply meta-mender-demo/recipes-core/psplash/psplash_git.bbappend]: @lluiscampos: Ok, maybe just stick to the original then. Although I cannot see the exact issue, it probably has to do with the fact that `_append` causes multiple entries to end up in the list. It's only for the demo, so it's not *that* important to fix it in an idiomatic way.

Comment (lluiscampos) [reply meta-mender-demo/recipes-core/psplash/psplash_git.bbappend:9]: Reverted to the original workaround. If the pipeline passes I'll merge.

**Comment by lluiscampos** [comment]
> > Running tests together with [mendersoftware/mender-qa#416](https://github.com/mendersoftware/mender-qa/pull/416) @ [Pipeline 195242424](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195242424)
> 
> New build @ [Pipeline 195338486](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195338486)

New build @ [Pipeline 195728117](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195728117)

**Comment by lluiscampos** [comment]
> > > Running tests together with [mendersoftware/mender-qa#416](https://github.com/mendersoftware/mender-qa/pull/416) @ [Pipeline 195242424](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195242424)
> > 
> > 
> > New build @ [Pipeline 195338486](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195338486)
> 
> New build @ [Pipeline 195728117](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195728117)

Rebased and yet another build @ [Pipeline 195940090](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/195940090)

**Comment by lluiscampos** [comment]
Reviewed at #1084, #1083.

**Comment by kacf** [comment]
@drewmoseley: Since some other changes may have made it into meta-mender since you started the work, you may want to rerun your layer checker to make sure we're not leaking any variables. Perhaps we should also think about how to test this.

**Pull request merged**